libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslFileUtil.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.ssl;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.AccessDeniedException;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.security.AccessControlException;
+import java.security.GeneralSecurityException;
+import java.security.UnrecoverableKeyException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Utility methods for common file handling in SSL configuration
+ */
+final class SslFileUtil {
+
+    static String pathsToString(List<Path> paths) {
+        return paths.stream()
+            .map(Path::toAbsolutePath)
+            .map(Object::toString)
+            .collect(Collectors.joining(","));
+    }
+
+    static SslConfigException ioException(String fileType, List<Path> paths, IOException cause) {
+        return ioException(fileType, paths, cause, null);
+    }
+
+    static SslConfigException ioException(String fileType, List<Path> paths, IOException cause, String detail) {
+        if (cause instanceof FileNotFoundException || cause instanceof NoSuchFileException) {
+            return fileNotFound(fileType, paths, cause);
+        }
+        if (cause instanceof AccessDeniedException) {
+            return accessDenied(fileType, paths, (AccessDeniedException) cause);
+        }
+        String message = "cannot read configured " + fileType;
+        if (paths.isEmpty() == false) {
+            message += " [" + pathsToString(paths) + "]";
+        }
+
+        if (cause.getCause() instanceof UnrecoverableKeyException) {
+            message += " - this is usually caused by an incorrect password";
+        } else if (cause != null && cause.getMessage() != null) {
```

Comment:
cause can't be null here, otherwise we'd have thrown an NPE in cause.getCause() above. We can check that cause is not null above and not go into this conditional at all